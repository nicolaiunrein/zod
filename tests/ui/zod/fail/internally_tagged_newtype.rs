#[derive(serde::Serialize, serde::Deserialize, zod::RequestType)]
#[zod(namespace = "Ns")]
#[serde(tag = "type")]
enum Test {
    Inner(u8),
}

#[derive(zod::Namespace)]
struct Ns;

fn main() {}
