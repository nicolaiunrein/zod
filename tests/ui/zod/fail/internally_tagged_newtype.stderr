error[E0080]: evaluation of constant value failed
 --> tests/ui/zod/fail/internally_tagged_newtype.rs:5:5
  |
5 |     Inner(u8),
  |     ^^^^^ the evaluated program panicked at 'zod: `internally tagged newtype variants are only supported for types compiling to zod objects.', $DIR/tests/ui/zod/fail/internally_tagged_newtype.rs:5:5
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant used
 --> tests/ui/zod/fail/internally_tagged_newtype.rs:5:5
  |
5 |     Inner(u8),
  |     ^^^^^

note: erroneous constant used
 --> tests/ui/zod/fail/internally_tagged_newtype.rs:1:48
  |
1 | #[derive(serde::Serialize, serde::Deserialize, zod::RequestType)]
  |                                                ^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the derive macro `zod::RequestType` (in Nightly builds, run with -Z macro-backtrace for more info)
